tests/test_commands.py

```diff
@@ -262,3 +262,37 @@ def test_js_comments(self):
         cmd.process(sql)
         self.assertEqual(1, cmd._exec_and_print.call_count)
         self.assertIn("CREATE FUNCTION", cmd._exec_and_print.mock_calls[0].args[0])
+
+
+class MultilineInputTest(TestCase):
+
+    def test_multiline_sql(self):
+        cmd = CrateShell()
+        cmd._exec_and_print = MagicMock()
+        cmd.process("SELECT\n1;\nSELECT 2;")
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT\n1;"),
+            call("SELECT 2;")
+        ])
+
+    def test_multiinput_multiline_sql(self):
+        cmd = CrateShell()
+        cmd._exec_and_print = MagicMock()
+        cmd.process_iterable(["SELECT\n1;\nSELECT 2;", "SELECT 3;"])
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT\n1;"),
+            call("SELECT 2;"),
+            call("SELECT 3;"),
+
+        ])
+
+    def test_input_commands_amd_sql(self):
+        cmd = CrateShell()
+        cmd._try_exec_cmd = MagicMock()
+        cmd._exec_and_print = MagicMock()
+        cmd.process_iterable(["SELECT 1;\n\\?", "SELECT 2;"])
+        cmd._try_exec_cmd.assert_called_once_with("?")
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT 1;"),
+            call("SELECT 2;"),
+        ])
```

Comment:
I've verified the command-line examples at GH-349 start working with your patch, both when using newlines for separating SQL statements, or not. Excellent.

```console
echo -e "select 1; select 2;" | crash
echo -e "select 1;\nselect 2;" | crash
```

Can you make sure you are also exercising both variants within your test cases?

Reply:
I have just added test cases to encompass the example you gave. I also improved the naming of existing test cases and added a few new ones. As a side note, I'm always testing with 3 statements instead of 2, because in my experience, a lot of things work with <= 2 elements, but not with more. :pain_emoji_needed: